lib/service/storage/elasticsearch.js

```diff
@@ -822,6 +822,79 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+   * Delete fields of a document and replace it
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Array}  fields - Document fields
+   * @param {Object} options - refresh (undefined), userId (null)
+   *
+   * @returns {Promise.<{ _id, _version, _source }>}
+   */
+  deleteFields (
+    index,
+    collection,
+    id,
+    fields,
+    { refresh, userId=null } = {})
+  {
+    const
+      esIndex = this._getESIndex(index, collection),
+      esRequest = {
+        id,
+        index: esIndex,
+      };
+
+    // metadata
+    const _kuzzle_info = {
+      author: getUserId(userId),
+      updatedAt: Date.now(),
+      updater: getUserId(userId)
+    };
+
+    return this._client.exists({ id, index: esIndex })
+      .then(({ body: exists }) => {
+        if (!exists) {
+          throw kerror.get('not_found', id, index, collection);
+        }
+
+        debug('DeleteFields document: %o', esRequest);
+        return this._client.get(esRequest)
+          .then(({ body }) => {
+            for (let field of fields) {
+              _.unset(body._source, field);
```

Comment:
You should set the fields values to `undefined` instead of deleting them
```suggestion
              _.set(body._source, field, undefined);
```